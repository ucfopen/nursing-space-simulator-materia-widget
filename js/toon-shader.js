AFRAME.registerShader('toon-shader', {
  schema: {
    "uDirLightPos": {type: 'vec3', default: '100 50 5', is: 'uniform' },
    "uDirLightColor": { type: 'color', default: 'black', is: 'uniform' },

    "uMaterialColor": { type: 'color', default: 'white', is: 'uniform' },

    "uKd": { type: "number", default: 0.7, is: "uniform" },
    "uBorder": { type: "number", default: 0.4, is: "uniform" }

  },
  vertexShader: [
    "varying vec3 vNormal;",
    "varying vec3 vViewPosition;",

    "void main() {",

      "gl_Position = projectionMatrix * modelViewMatrix * vec4( position, 1.0 );",
      "vNormal = normalize( normalMatrix * normal );",
      "vec4 mvPosition = modelViewMatrix * vec4( position, 1.0 );",
      "vViewPosition = -mvPosition.xyz;",

    "}"

  ].join("\n"),

  fragmentShader: [   
      "uniform vec3 uMaterialColor;",

      "uniform vec3 uDirLightPos;",
      "uniform vec3 uDirLightColor;",

      "uniform float uKd;",
      "uniform float uBorder;",

      "varying vec3 vNormal;",
      "varying vec3 vViewPosition;",

      "void main() {",

        // compute direction to light
        "vec4 lDirection = viewMatrix * vec4( uDirLightPos, 0.0 );",
        "vec3 lVector = normalize( lDirection.xyz );",

        // diffuse: N * L. Normal must be normalized, since it's interpolated.
        "vec3 normal = normalize( vNormal );",
        //was: "float diffuse = max( dot( normal, lVector ), 0.0);",
        // solution
        "float diffuse = dot( normal, lVector );",
        "if ( diffuse > 0.6 ) { diffuse = 1.0; }",
        "else if ( diffuse > -0.2 ) { diffuse = 0.7; }",
        "else { diffuse = 0.3; }",

        "gl_FragColor = vec4( uKd * uMaterialColor * uDirLightColor * diffuse, 1.0 );",

      "}"

    ].join("\n")
});