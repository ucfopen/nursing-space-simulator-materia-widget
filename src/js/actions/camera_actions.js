export const CAMERA_UPDATE_POSITION = "CAMERA_UPDATE_POSITION";
export const CAMERA_TOGGLE = "CAMERA_TOGGLE_TYPE";

export function updateCameraPosition(axisDirection) {
	return {
		type: CAMERA_UPDATE_POSITION,
		payload: axisDirection
	};
}

export function toggleCameraType() {
	return {
		type: CAMERA_TOGGLE
	};
}
