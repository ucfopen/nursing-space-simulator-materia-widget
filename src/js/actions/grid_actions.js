export const DELETE_ASSET = "DELETE_ASSET";
export const DESELECT_ASSET = "DESELECT_ASSET";
export const INSERT_ASSET = "INSERT_ASSET";
export const PLACE_ASSET = "PLACE_ASSET";
export const ROTATE_ASSET = "ROTATE_ASSET";
export const SELECT_ASSET = "SELECT_ASSET";
export const SELECT_ASSET_TYPE = "SELECT_ASSET_TYPE";
export const REMOVE_ASSET = "REMOVE_ASSET";
export const UPDATE_ASSET_POSITION = "UPDATE_ASSET_POSITION";
export const EXTEND_WALL = "EXTEND_WALL";
export const FILL_WALLS = "FILL_WALLS";

export function selectAssetType(asset) {
	return {
		type: SELECT_ASSET_TYPE,
		payload: asset
	};
}

export function selectAsset(asset, x, z) {
	return {
		type: SELECT_ASSET,
		payload: { asset, x, z }
	};
}

export function deselectAsset() {
	return {
		type: DESELECT_ASSET
	};
}

export function rotateAsset(x, z) {
	return {
		type: ROTATE_ASSET,
		payload: { x, z }
	};
}

export function removeAsset(x, z) {
	return {
		type: REMOVE_ASSET,
		payload: { x, z }
	};
}

export function insertAsset(x, z, assetId = null) {
	return {
		type: INSERT_ASSET,
		payload: { assetId, x, z }
	};
}

export function updateAssetPosition(axisDirection) {
	return {
		type: UPDATE_ASSET_POSITION,
		payload: axisDirection
	};
}

export function extendWall(x, z) {
	return {
		type: EXTEND_WALL,
		payload: { x, z }
	};
}

export function fillWalls(x, z, assetId = null, extendX, extendZ, validX, validZ) {
	return {
		type: FILL_WALLS,
		payload: { assetId, x, z, extendX, extendZ, validX, validZ }
	};
}
