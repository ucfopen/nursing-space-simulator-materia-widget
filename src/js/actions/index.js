import { loadGrid } from "../grid";

export const INIT_DATA = "init_data";

export function initData(qset) {
	const grid = loadGrid(
		qset.options.gridLoader.content,
		qset.options.gridLoader.rows,
		qset.options.gridLoader.columns
	);

	const categories = qset.options.categories;
	const stickerTypes = qset.options.stickerTypes;
	const assets = qset.options.assets;

	return {
		type: INIT_DATA,
		payload: { grid, categories, stickerTypes, assets }
	};
}
