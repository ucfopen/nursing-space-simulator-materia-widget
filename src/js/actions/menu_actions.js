export const TOGGLE_MENU_VISIBILITY = "toggle_menu_visibility"
export const SET_CATEGORY = "set_menu_category"

export function toggleMenuVisibility() {
  return {
    type: TOGGLE_MENU_VISIBILITY
  }
}

export function setCategory(category) {
  return {
    type: SET_CATEGORY,
    payload: category
  }
}
