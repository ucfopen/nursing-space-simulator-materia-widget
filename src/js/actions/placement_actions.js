export const SELECT_ASSET_TYPE = "select_asset_type"
export const UPDATE_GRID = "update_grid"
export const UPDATE_GRID_NULL = "update_grid_null"
export const SELECT_OBJECT = "select_object"
export const DESELECT_OBJECT = "deselect_object"
export const ROTATE_OBJECT = "rotate_object"
export const REMOVE_OBJECT = "remove_object"

export function selectAssetType(asset) {
  return {
    type: SELECT_ASSET_TYPE,
    payload: asset
  }
}

export function selectObject(asset, x, y) {
  return {
    type: SELECT_OBJECT,
    payload: { asset, x, y }
  }
}

export function deselectObject() {
  return {
    type: DESELECT_OBJECT
  }
}

export function rotateObject(asset, x, y) {
  return {
    type: ROTATE_OBJECT,
    payload: { asset, x, y }
  }
}

export function removeObject(x, y) {
  return {
    type: REMOVE_OBJECT
  }
}

export function updateGrid(x, y, asset) {
  if (asset)
    return {
      type: UPDATE_GRID,
      payload: {
        x,
        y,
        asset
      }
    }
  else
    return {
      type: UPDATE_GRID_NULL
    }
}
