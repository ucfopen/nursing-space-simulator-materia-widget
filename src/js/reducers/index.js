import { combineReducers } from "redux";
import Position from "./positionReducer";
import Menu from "./menuReducer";
import Grid from "./gridReducer";
import Tour from "./tourReducer";

const rootReducer = combineReducers({
	grid: Grid,
	menu: Menu,
	position: Position,
	tour: Tour
});

export default rootReducer;
