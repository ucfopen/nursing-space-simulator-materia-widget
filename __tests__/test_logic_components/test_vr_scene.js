import ConnectedVRScene, { VRScene } from "../../src/js/components/vr_scene";
import React from "react";
import { Provider } from "react-redux";
import { mount, shallow } from "enzyme";
import { createStore } from "redux";
import renderer from "react-test-renderer";
import reducers from "../../src/js/reducers";
import { initData } from "../../src/js/actions/";
import qset from "../custom_test_utilities/mock_qset";
import { jsdom } from "jsdom";

const position = { x: 0, y: 0, z: 0 };
const assets = {
	bed: {
		id: "bed-1",
		category: "beds",
		objSrc: "url/to/obj",
		mtlSrc: "url/to/mtl"
	},
	chair: {
		id: "chair",
		category: "equipment",
		objSrc: "url/to/obj",
		mtlSrc: "url/to/mtl"
	},
	wall: {
		id: "chair",
		category: "equipment"
	}
};
const grid = [
	["0", "0", "0"],
	["0", { id: "bed", rotation: 180 }, "0"],
	["0", "0", "0"]
];

const propsInVRScene = [
	"qset",
	"position",
	"thirdPerson",
	"assets",
	"grid",
	"currentX",
	"currentY",
	"insertAsset",
	"selectAsset",
	"store",
	"storeSubscription"
];

const insertAsset = jest.fn();
const selectAsset = jest.fn();

describe("VR Scene Tests", () => {
	it("should display loading if no props are passed in", () => {
		const vrSceneTree = renderer.create(<VRScene />).toJSON();
		expect(vrSceneTree).toMatchSnapshot();
	});

	it("should display loading when no position is passed in", () => {
		const vrSceneTree = renderer
			.create(<VRScene grid={grid} assets={assets} />)
			.toJSON();
		expect(vrSceneTree).toMatchSnapshot();
	});

	it("should display loading when no assets are passed in", () => {
		const vrSceneTree = renderer
			.create(<VRScene grid={grid} position={position} />)
			.toJSON();
		expect(vrSceneTree).toMatchSnapshot();
	});

	it("should display the scene when all props are passed in", () => {
		const vrSceneTree = renderer
			.create(
				<VRScene
					grid={grid}
					assets={assets}
					position={position}
					selectAsset={selectAsset}
					insertAsset={insertAsset}
					currentX={0}
					currentY={0}
					thirdPerson={false}
				/>
			)
			.toJSON();
		expect(vrSceneTree).toMatchSnapshot();
	});

	it("should not render ceiling when in third person", () => {
		const vrSceneTree = renderer
			.create(
				<VRScene
					grid={grid}
					assets={assets}
					position={position}
					selectAsset={selectAsset}
					insertAsset={insertAsset}
					currentX={1}
					currentY={1}
					thirdPerson={true}
				/>
			)
			.toJSON();
		expect(vrSceneTree).toMatchSnapshot();
	});

	it("should render connected VRScene with props", () => {
		const store = createStore(reducers);

		// A manual dispatch is required since a single component is being tested
		store.dispatch(initData(qset));

		const connectedVRSceneComponent = shallow(
			<ConnectedVRScene store={store} qset={qset} />
		);

		const propsToBeTested = connectedVRSceneComponent.props();

		Object.keys(propsToBeTested).map(prop => {
			expect(propsInVRScene.includes(prop)).toBe(true);
			expect(propsToBeTested[prop]).not.toBe(undefined);
		});
	});
});
