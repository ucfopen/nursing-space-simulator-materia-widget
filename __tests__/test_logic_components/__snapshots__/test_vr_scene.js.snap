// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`VR Scene Tests should display loading if no props are passed in 1`] = `
<div>
  Loading
</div>
`;

exports[`VR Scene Tests should display loading when no assets are passed in 1`] = `
<div>
  Loading
</div>
`;

exports[`VR Scene Tests should display loading when no position is passed in 1`] = `
<div>
  Loading
</div>
`;

exports[`VR Scene Tests should display the scene when all props are passed in 1`] = `
<a-scene
  className="vr-scene"
>
  <a-assets>
    <img
      alt="ceiling-texture"
      id="ceilingTexture"
      src="assets/CEILING_TILE.jpg"
    />
    <a-asset-item
      id="bed-obj"
      src="url/to/obj"
    />
    <a-asset-item
      id="chair-obj"
      src="url/to/obj"
    />
    <a-asset-item
      id="bed-mtl"
      src="url/to/mtl"
    />
    <a-asset-item
      id="chair-mtl"
      src="url/to/mtl"
    />
  </a-assets>
  <a-sky />
  <a-plane />
  <a-plane />
  <a-entity />
  <a-entity />
  <a-plane />
  <a-plane />
  <a-plane />
  <a-plane />
  <a-plane />
  <a-plane />
  <a-plane />
  <a-plane />
  <a-plane />
</a-scene>
`;

exports[`VR Scene Tests should not render ceiling when in third person 1`] = `
<a-scene
  className="vr-scene"
>
  <a-assets>
    <img
      alt="ceiling-texture"
      id="ceilingTexture"
      src="assets/CEILING_TILE.jpg"
    />
    <a-asset-item
      id="bed-obj"
      src="url/to/obj"
    />
    <a-asset-item
      id="chair-obj"
      src="url/to/obj"
    />
    <a-asset-item
      id="bed-mtl"
      src="url/to/mtl"
    />
    <a-asset-item
      id="chair-mtl"
      src="url/to/mtl"
    />
  </a-assets>
  <a-sky />
  <a-plane />
  <a-entity />
  <a-entity />
  <a-plane />
  <a-plane />
  <a-plane />
  <a-plane />
  <a-plane />
  <a-plane />
  <a-plane />
  <a-plane />
  <a-plane />
</a-scene>
`;

exports[`VR Scene Tests should render floor correctly if a selected asset is passed in 1`] = `
<a-scene
  className="vr-scene"
>
  <a-assets>
    <img
      alt="ceiling-texture"
      id="ceilingTexture"
      src="assets/CEILING_TILE.jpg"
    />
    <a-asset-item
      id="bed-obj"
      src="url/to/obj"
    />
    <a-asset-item
      id="chair-obj"
      src="url/to/obj"
    />
    <a-asset-item
      id="bed-mtl"
      src="url/to/mtl"
    />
    <a-asset-item
      id="chair-mtl"
      src="url/to/mtl"
    />
  </a-assets>
  <a-sky />
  <a-plane />
  <a-entity />
  <a-entity />
  <a-plane />
  <a-plane />
  <a-plane />
  <a-plane />
  <a-plane />
  <a-plane />
  <a-plane />
  <a-plane />
  <a-plane />
</a-scene>
`;
